specutils/analysis/width.py

```diff
@@ -0,0 +1,212 @@
+import numpy as np
+from astropy.stats.funcs import gaussian_fwhm_to_sigma
+from ..spectra import SpectralRegion
+
+
+__all__ = ['gaussian_sigma_width', 'gaussian_fwhm', 'fwhm']
+
+
+def _computation_wrapper(func, spectrum, region):
+
+    # No region, therefore whole spectrum.
+    if region is None:
+        return func(spectrum)
+
+    # Single region
+    elif isinstance(region, SpectralRegion):
+        return func(spectrum, region=region)
+
+    # List of regions
+    elif isinstance(region, list):
+        return [func(spectrum, region=reg) for reg in region]
+
+
+def gaussian_sigma_width(spectrum, region=None):
+    """
+    Estimate the full width of the spectrum based on an approximate value of
+    sigma.  This will be calculated over the regions, if they are specified.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
+        The spectrum object over which the width will be calculated.
+
+    region: `~specutils.utils.SpectralRegion` or list of `~specutils.utils.SpectralRegion`
+        Region within the spectrum to calculate the gaussian sigma width.
```

Comment:
Here and below, maybe add something like "If region is None, this will be calculated over the entire spectrum"?

Reply:
Sounds good. Btw, I wonder what you think about removing the docstrings for the private methods below. They mostly just repeat what the public functions say already.